lib/sign.js

```diff
@@ -17,11 +18,51 @@ var xpi = require("./xpi");
 
 var AMO_API_PREFIX = require("./settings").AMO_API_PREFIX;
 
+function getXpiInfo(options) {
+  options = _.assign({
+    addonDir: null,
+    xpiPath: null,
+  }, options);
+
+  if (!options.xpiPath) {
+    return utils.getManifest({addonDir: options.addonDir})
+      .then(function(manifest) {
+        return getXpiInfoFromManifest(manifest);
+      });
+  } else {
+    return utils.extractXPI(options.xpiPath)
```

Comment:
Actually `getXPIManifest` doesn't have anything specific to a `package.json` file in there. You could potentially load an `install.rdf` as a fallback there if all you need is data they both have.